pkg/lib/machines.js

```diff
@@ -591,10 +591,13 @@
                         whirl();
                     })
                 .on("close", function(ev, options) {
-                    problem = options.problem || "disconnected";
-                    open = false;
-                    state(host, "failed", problem);
                     var m = machines.lookup(host);
+                    open = false;
+                    if (!options.problem && m && !m.visible)
+                        // reset to clean state when removing machine (orderly disconnect)
+                        state(host, null, null);
```

Comment:
I think we put braces around multi-line blocks. Even if one of those lines is a comment, or line-wrapped line.